interp/api.go

```diff
@@ -366,28 +372,37 @@ func StdIO(in io.Reader, out, err io.Writer) RunnerOption {
 	}
 }
 
-func (r *Runner) optByName(name string, bash bool) *bool {
+func (r *Runner) optByName(name string, bash bool) (*bool, *shellOpt, *bashOpt) {
```

Comment:
is the second result parameter ever used?

perhaps as a way to clarify what they mean: `(posix *bool, bash *bashOpt)`

Or, for consistency, using the struct for both: `(*shellOpt, *bashOpt)`